composer: allow Symfony 3.x

MacDada opened the pull request:
#8

MacDada committed:
composer: allow Symfony 3.x

https://github.com/dantleech/symfony-form-array-to-delimited-string-transformer/issues/8

dantleech commented:
hey, the build is failing because PHP 5.5 doesn't seem to be available on travis anymore, would be great if you could update, otherwise I can have a look when I have 10 minutes.

MacDada committed:
travis build: upgraded php and phpunit

https://github.com/dantleech/symfony-form-array-to-delimited-string-transformer/pull/9#issuecomment-670840905

MacDada commented:
> hey, the build is failing because PHP 5.5 doesn't seem to be available on travis anymore, would be great if you could update (maybe just use PHP 7.0), otherwise I can have a look when I have 10 minutes.

done

dantleech commented:
Thanks for that - would ultimately be good to add support for 4.x and 5.x too and update Travis accordingly, but I guess we can do that if and when it's required ...

dantleech commented:
Also should probably drop 5.3 support and start support at f.e. `7.1`

MacDada commented on tests/ArrayToDelimitedStringTransformerTest.php:
```diff
@@ -114,7 +116,8 @@ public function provideTransform()
     {
         if ($expectedException) {
-            $this->setExpectedException('Symfony\Component\Form\Exception\TransformationFailedException', $expectedException);
+            $this->expectException('Symfony\Component\Form\Exception\TransformationFailedException');
```
> Could replace with `TransformationFailedException::class`

Yeah, I didn't know if you want to bump php version requirement for the lib.

MacDada commented:
> Thanks for that - would ultimately be good to add support for 4.x and 5.x too and update Travis accordingly, but I guess we can do that if and when it's required ...

I might need that next week, haha
